secure.go

```diff
@@ -232,3 +268,47 @@ func (s *Secure) Process(w http.ResponseWriter, r *http.Request) error {
 
 	return nil
 }
+
+// Nonce returns the nonce value associated with the present request. If no nonce has been generated it returns an empty string.
+func Nonce(r *http.Request) string {
```

Comment:
I'd prefer if this took a context.Context instead of a *http.Request.

Couple of reasons:
a) If a sub-function performs the rendering, and needs the CSP nonce, it is more likely to already have a context.Context argument, than a *http.Request argument.
b) This mechanism is a bit more general than HTTP. For instance, if my request handler renders HTML into a gRPC response (for instance, if the main HTTP handler makes an RPC to a microservice to render a piece of the page), it can use the same mechanism to retrieve the nonce, even though its not (explicitly) inside an HTTP transaction. (see also other request about exposing a WithCSPNonce function).